Yafc.Parser/Data/FactorioDataDeserializer.cs

```diff
@@ -130,9 +130,13 @@ public Project LoadData(string projectPath, LuaTable data, LuaTable prototypes,
         DeserializePrototypes(raw, "recipe", DeserializeRecipe, progress, errorCollector);
         progress.Report(("Loading", "Loading technologies"));
         DeserializePrototypes(raw, "technology", DeserializeTechnology, progress, errorCollector);
-        progress.Report(("Loading", "Loading entities"));
-        LuaTable entityPrototypes = (LuaTable?)prototypes["entity"] ?? throw new ArgumentException("Could not load prototypes.item from data argument", nameof(prototypes));
+        progress.Report(("Loading", "Loading qualities"));
+        DeserializePrototypes(raw, "quality", DeserializeQuality, progress, errorCollector);
+        Quality.Normal = GetObject<Quality>("normal");
+        rootAccessible.Add(Quality.Normal);
```

Comment:
Should this be protected with a check to be sure the `quality` module is loaded/available? (With some default if not?)
So we won't run into issues to support if base game (when the user doesn't have the DLC)

Reply:
The base game defines the qualities normal (in base) and unknown-quality (in core). I've loaded Yafc several times without the quality mod enabled.

It looks like modpacks that don't define a normal quality will have good-enough behavior; a quick check says the quality features in Yafc will be disabled, but everything else will work. (I changed `GetObject<Quality>("normal")` to `GetObject<Quality>("normal2")`. It created a default quality object with no icon and a null `nextQuality`.)

Reply:
Ah... so quality is basically build into the base game/mod. And the quality mod will make use of it to extend this feature into something usable?

This seems to make things nice and 'easy' for us. (I was a bit worried that we broke the compatibility with the non-DLC players (as we did a little white back)